logstash-core/src/main/java/org/logstash/config/ir/compiler/RubyIntegration.java

```diff
@@ -0,0 +1,98 @@
+package org.logstash.config.ir.compiler;
+
+import java.util.Collection;
+import org.jruby.RubyHash;
+import org.jruby.RubyInteger;
+import org.jruby.RubyString;
+import org.jruby.runtime.builtin.IRubyObject;
+import org.logstash.ext.JrubyEventExtLibrary;
+
+/**
+ * This class holds interfaces implemented by Ruby concrete classes.
+ */
+public final class RubyIntegration {
+
+    private RubyIntegration() {
+        //Utility Class.
+    }
+
+    /**
+     * A Ruby Plugin.
+     */
+    public interface Plugin {
+        void register();
+    }
+
+    /**
+     * A Ruby Filter. Currently, this interface is implemented only by the Ruby class
+     * {@code FilterDelegator}.
+     */
+    public interface Filter extends RubyIntegration.Plugin {
+
+        /**
+         * Same as {@code FilterDelegator}'s {@code multi_filter}.
+         * @param events Events to Filter
+         * @return Filtered {@link JrubyEventExtLibrary.RubyEvent}
+         */
+        Collection<JrubyEventExtLibrary.RubyEvent> multiFilter(
+            Collection<JrubyEventExtLibrary.RubyEvent> events
+        );
+
+        Collection<JrubyEventExtLibrary.RubyEvent> flush(RubyHash options);
+
+        /**
+         * Checks if this filter has a flush method.
+         * @return True iff this filter has a flush method
+         */
+        boolean hasFlush();
+
+        /**
+         * Checks if this filter does periodic flushing.
+         * @return True iff this filter uses periodic flushing
+         */
+        boolean periodicFlush();
+    }
+
+    /**
+     * A Ruby Output. Currently, this interface is implemented only by the Ruby class
+     * {@code OutputDelegator}.
+     */
+    public interface Output extends RubyIntegration.Plugin {
+        void multiReceive(Collection<JrubyEventExtLibrary.RubyEvent> events);
+    }
+
+    /**
+     * The Main Ruby Pipeline Class. Currently, this interface is implemented only by the Ruby class
+     * {@code BasePipeline}.
+     */
+    public interface Pipeline {
+
+        IRubyObject buildInput(RubyString name, RubyInteger line, RubyInteger column,
```

Comment:
IRubyObject is probably too generic. Any reason not to add an interface for Input similar to Output and Filter?

Additionally, can we move away from JRuby objects (RubyString, etc) in the Java interfaces? The reason I ask is that I am working on the Java Plugin API which will be pure Java (no JRuby) and I'm wondering about the difficulty of moving away from IRubyObject, RubyString, etc, to support the future Java Plugin API.

Reply:
@jordansissel currently I have no use for a Java `Input` here as explained above. Let's not add more code than we have to until we need it? :)

Reply:
@jordansissel also if we want a Java typed interface here, we can just add an overload that does the conversion when we need it. But until then I see no reason adding redundant code steps in, right? Also, I'd just naturally remove these once we ported the functions behind these to Java preferably instead of adding a million wrappers that no one understands :)